DDG4/python/DDSim/DD4hepSimulation.py

```diff
@@ -467,31 +411,134 @@ def run(self):
 
     self.part.setupUserParticleHandler(part, kernel, DDG4)
 
-    # =================================================================================
+    return 1
+
+  def __setupSensitives(geant4, detectorDescription, sim):
+    kernel = geant4.kernel()
 
     # Setup global filters for use in sensitive detectors
     try:
-      self.filter.setupFilters(kernel)
+      sim.filter.setupFilters(kernel)
     except RuntimeError as e:
       logger.error("%s", e)
       exit(1)
 
-    # =================================================================================
     # get lists of trackers and calorimeters in detectorDescription
-
-    trk, cal, unk = self.getDetectorLists(detectorDescription)
+    trk, cal, unk = sim.getDetectorLists(detectorDescription)
     for detectors, function, defFilter, defAction, abort in \
-        [(trk, geant4.setupTracker, self.filter.tracker, self.action.tracker, False),
-         (cal, geant4.setupCalorimeter, self.filter.calo, self.action.calo, False),
+        [(trk, geant4.setupTracker, sim.filter.tracker, sim.action.tracker, False),
+         (cal, geant4.setupCalorimeter, sim.filter.calo, sim.action.calo, False),
          (unk, geant4.setupDetector, None, "No Default", True),
          ]:
       try:
-        self.__setupSensitiveDetectors(detectors, function, defFilter, defAction, abort)
+        sim.__setupSensitiveDetectors(detectors, function, defFilter, defAction, abort)
       except Exception as e:
         logger.error("Failed setting up sensitive detector %s", e)
         raise
 
-  # =================================================================================
+    return 1
+
+  def __setupWorker(geant4, sim):
```

Comment:
As is this is still an instance method.
```suggestion
  def __setupWorker(self, geant4)
```
and accordingly below? Or why would this not work?
Then you either need to move out of the class or add the `@staticmethod` decorator.
And is the return value required by the interface?

Reply:
I can (and should) likely make this work with `@staticmethod`. The return value appears to be unused. I based the structure on what is done in `SiDSim_MT.py`, to the extent it provides any prior art.

Reply:
(I was unable to make this work with an instance method since I suspect that by the time `__setupWorker` is called, the DD4hepSimulation object is not available to the thread that calls `__setupWorker`.)

Reply:
But according to the indentation, this still _is_ an instance method? Just because the first argument isn't `self` doesn't change that, does it?

Reply:
Made this into an instance method in 1b409705f501fc3c106619b4e988956d2b29b845.